tfx/components/example_gen/base_example_gen_executor.py

```diff
@@ -44,10 +45,24 @@
 EXAMPLES_KEY = 'examples'
 
 
-def _PartitionFn(record: bytes, num_partitions: int, buckets: List[int]) -> int:
+def _PartitionFn(record: bytes, num_partitions: int, buckets: List[int], 
+                 split_config: example_gen_pb2.SplitConfig) -> int:
   assert num_partitions == len(
       buckets), 'Partitions do not match bucket number.'
-  bucket = int(hashlib.sha256(record).hexdigest(), 16) % buckets[-1]
+  partition_str = record
+  if split_config.HasField("partition_feature_name"):
+    # Use a feature for partitioning the examples.
+    feature_name = split_config.partition_feature_name
+    # Deserialize the record to tf.train.Example. 
+    example = tf.train.Example()
+    example.ParseFromString(record)
+    feature = example.features.feature[feature_name]
```

Comment:
here we need to validate whether the feature mentioned exists in the example. 

please also add a test to guard the behavior of this invalid case

Reply:
got it, thanks for the catch